### ISSUE TITLE
Possibility of implementing new engines

### ISSUE THREAD

**Issue by matbrgz** [duolingo/metasearch#7]
Is the project open to receive new engine implementations?

I thought about adding a feature flag to enable / disable engines.

**Comment by artnc** [comment]
Yes, PRs are welcome! Users can disable an engine by simply deleting it from `config.yaml` if that's what you mean.

**Comment by guillim** [comment]
Hello, very interesting project !!!

Is there somewhere we can vote for new search engines ?

I am thinking gmail, Outlook and other mail services could be 👍🏻

**Comment by artnc** [comment]
@guillim It's a bit hard for me to find time to set up a test account, populate it with dummy data, write the engine code etc for services that we don't actually use or plan to expose internally here at Duolingo, but it should be straightforward to [add](https://github.com/duolingo/metasearch/pull/11) [support](https://github.com/duolingo/metasearch/pull/13) yourself for any service you'd like! To date I've never vetoed such a PR :)

**Comment by jwpjrdev** [comment]
Adding onto this, an email engine would be real dandy.

**Comment by jwpjrdev** [comment]
Google or Apple Calendar would also be super useful

**Comment by hKQwHW** [comment]
Would it be possible to add support for images on Trello and Nuclino? I've already tried implementing these, but I am not good at writing TypeScript, unfortunately.

**Comment by artnc** [comment]
I've never used either of those (@jimmatthews was the one who added [Trello support](https://github.com/duolingo/metasearch/pull/11)) but certainly happy to accept PRs from anyone for those features

**Comment by hKQwHW** [comment]
Alright, got it. If I can figure out how to work with Typescript better, I will make a PR. I've always used vanilla JS/HTML/CSS.

**Comment by roots-ai** [comment]
https://github.com/duolingo/metasearch/issues/32
https://github.com/duolingo/metasearch/issues/33

**Comment by hverlin** [comment]
Cool project! You are welcome to reuse anything from [Quest](https://github.com/hverlin/Quest), (e.g. code to fetch mails from [gmail](https://github.com/hverlin/Quest/blob/master/src/modules/gmail/index.js#L240))
Similar idea, but not enough time and motivation to maintain it.
